### PR TITLE
refactor(proposer): background proposer/da submitter logic

### LINKED ISSUE #2875: calculate_gas in testnet  need too much
### Rooch version

rooch 0.7.6 (git commit e5ba2429862ad18b84e283cffa13ac99b49a71d7)

### What did you do?

rooch move run --function $PKG::gold_miner::start --args "object:$TREASURY" --args "address:$SENDER"

### What did you see happen?

Transaction error: ErrorObject { code: ServerError(2), message: "status ABORTED of type Execution with sub status 1004", data: None }

### What did you expect to see?

```move
 // === validate gas ===
        let max_gas_amount = tx_context::max_gas_amount();
        let gas = transaction_fee::calculate_gas(max_gas_amount);

        let gas_schedule = gas_schedule::gas_schedule();
        let max_gas_amount_config = gas_schedule::gas_schedule_max_gas_amount(gas_schedule);
        assert!(
            max_gas_amount <= max_gas_amount_config,
            auth_validator::error_validate_max_gas_amount_exceeded(),
        );

        let gas_balance = gas_coin::balance(sender);

        // we do not need to check the gas balance in local or dev chain
        if(!chain_id::is_local_or_dev()){
            assert!(
                gas_balance >= gas,
                auth_validator::error_validate_cant_pay_gas_deposit(),
            );
        };
```

我只领取一次gas: 钱包大概0.97 RGAS左右。

无法正常调用程序。计算过程的gas要求太高。

重新多领取几次水之后可以正常调用了。

### PR DESCRIPTION
**Pull request by popcnt1**
## Summary

1. add proposer store to manage last block state: Introduces `ProposerStore` and `ProposerDBStore` implementations for managing the proposer last block state, including get, set, and clear operations. Integrates the proposer store into the `RoochStore` structure.
2. add transaction batching mechanism: Introduced a new transaction batching mechanism to the Rooch DA server. Added `BatchMaker` to handle batch intervals based on transaction orders and timestamps. Enhanced the `DAServerActor` to support appending transactions and updated relevant modules to incorporate new functionality.
3. add descriptive message for unsupported column family operations and flush on drop_cf
4. add ProposerConfig struct with serialization and command-line support
5. optimize SCC block proposal
    
    - Removed the proxy module and associated actor messages for proposing transactions.
    - Refactored StateCommitmentChain to integrate directly with MoveOSStore and RoochStore for block proposals.
    - Improved block proposal logic to handle state roots and accumulator roots accurately.
    - Updated ProposerActor initialization to set last proposed block based on configuration.
6. remove PutDABatchMessage and related submission logic

Close #2875 : make batch interval contronled by settings, default is 15mins

### CODE REVIEW

File: crates/rooch-types/src/da/state.rs [hunk 1]
```diff
@@ -15,8 +15,10 @@ pub struct DAServerStatus {
     /// Should be closed to request time if there were new blocks.
     /// None if no blocks were received after server start.
     pub last_block_update_time: Option<u64>,
+    /// Min available block number. if not set, all blocks will be submitted to DA.
+    pub min_avail_block_number: Option<u128>,
```
Comment (jolestar) [inline crates/rooch-types/src/da/state.rs:19]: 这是一个配置项，似乎没必要输出在状态里？

File: crates/rooch-da/src/actor/server.rs [hunk 2]
```diff
@@ -196,51 +198,112 @@ impl DAServerActor {
...
-            .await?;
-        self.last_block_number = Some(block_number);
-        self.last_block_update_time = SystemTime::now()
-            .duration_since(time::UNIX_EPOCH)?
-            .as_secs();
+        msg: AppendTransactionMessage,
+    ) -> anyhow::Result<()> {
+        let tx_order = msg.tx_order;
+        let tx_timestamp = msg.tx_timestamp;
+        let batch = self.batch_maker.append_transaction(tx_order, tx_timestamp);
```
Comment (jolestar) [inline crates/rooch-da/src/actor/server.rs:215]: 如果节点重启了，是否会影响 batch_maker ？

File: crates/rooch-types/src/da/state.rs (hunk 1 above)
Comment (popcnt1) [reply crates/rooch-types/src/da/state.rs:19]: 放这里主要用来解释可用block的区间，直接在注释里说明一下也行

File: crates/rooch-da/src/actor/server.rs (hunk 2 above)
Comment (popcnt1) [reply crates/rooch-da/src/actor/server.rs:215]: 节点重启，da store 会在启动时自动根据lastorder（sequencer)来补全block，不会有tx丢失

**Review by jolestar — APPROVED** [review]

**Pull request merged**